include/deal.II/base/quadrature_lib.h

```diff
@@ -511,6 +511,111 @@ class QTelles: public Quadrature<dim>
+*
+* @author Giuseppe Pitton, Luca Heltai 2015
+**/
+template <int dim>
+class QGaussChebyshev : public Quadrature<dim>
+{
+public:
+  //Generate a formula with <tt>n</tt> quadrature points
+  QGaussChebyshev(const unsigned int n);
+
+protected:
+  // Sets the points of the quadrature formula.
+  std::vector<double>
+  set_quadrature_points(const unsigned int n) const;
+
+  // Sets the weights of the quadrature formula.
+  std::vector<double>
+  set_quadrature_weights(const unsigned int n) const;
+
+};
+
+
+/**
+* Gauss-Radau-Chebyshev quadrature rules integrate the weighted product
+* $\int_{-1}^1 f(x) w(x) dx$ with weight given by:
+* $w(x) = 1/\sqrt{1-x^2}$ with the additional constraint that a quadrature point
+* lies at one of the two extrema of the interval.
+* The nodes and weights are known analytically,
+* and are exact for monomials up to the order $2n-2$, where $n$ is the number
+* of quadrature points.
+* Here we rescale the quadrature formula so that it is defined on t
+* he interval [0,1] instead of [-1,1]. So the quadrature formulas inte
+* grate exactly the integral $\int_0^1 f(x) w(x) dx$ with the weight:
+* $w(x) = 1/sqrt{x(1-x)}$. By default the quadrature is constructed with the
+* left endpoint as quadrature node, but the quadrature node can be imposed at the
+* right endpoint through the variable ep that can assume the values left or right.
+*
+* @author Giuseppe Pitton, Luca Heltai 2015
+**/
+template <int dim>
+class QGaussRadauChebyshev : public Quadrature<dim>
+{
+public:
+  enum EndPoint { left,right };
+  EndPoint ep;
+  //Generate a formula with <tt>n</tt> quadrature points
+  QGaussRadauChebyshev(const unsigned int n,
+                       QGaussRadauChebyshev::EndPoint ep=QGaussRadauChebyshev::left);
+
+protected:
+  // Sets the points of the quadrature formula.
+  std::vector<double>
+  set_quadrature_points(const unsigned int n) const;
+
+  // Sets the weights of the quadrature formula.
+  std::vector<double>
+  set_quadrature_weights(const unsigned int n) const;
+
+};
+
+/**
+* Gauss-Lobatto-Chebyshev quadrature rules integrate the weighted product
+* $\int_{-1}^1 f(x) w(x) dx$ with weight given by:
+* $w(x) = 1/\sqrt{1-x^2}$. The nodes and weights are known analytically,
+* and are exact for monomials up to the order $2n-1$, where $n$ is the number
+* of quadrature points.
+* Here we rescale the quadrature formula so that it is defined on t
+* he interval [0,1] instead of [-1,1]. So the quadrature formulas inte
+* grate exactly the integral $\int_0^1 f(x) w(x) dx$ with the weight:
+* $w(x) = 1/sqrt{x(1-x)}$.
+* For details see:
+* M. Abramowitz & I.A. Stegun: Handbook of Mathematical Functions, par. 25.4.40
+*
+* @author Giuseppe Pitton, Luca Heltai 2015
+**/
+template <int dim>
+class QGaussLobattoChebyshev : public Quadrature<dim>
+{
+public:
+  //Generate a formula with <tt>n</tt> quadrature points
+  QGaussLobattoChebyshev(const unsigned int n);
+
+protected:
```

Comment:
Here and in the other classes, I think you can make the two member functions `private` instead of `protected`.

Also, the functions really are misnamed as they do not *set* anything. They just compute something and then return it. How about `get_quadrature_points/weights()`? I believe that you can probably make them `static` as well, since they don't depend on any of the member variables.